rust/tools/authorization-logic/src/souffle/lowering_ast_datalog.rs

```diff
@@ -87,15 +87,15 @@
 //! Queries check of a single unconditional assertion is true. A query of
 //! the form:
 //!
-//!      Q_NAME = query ASSERTION(<ARGS>) ?
+//!      // Q_NAME = query ASSERTION(<ARGS>) ?
 //!
 //! is translated into an assertion with just one argument, and this
 //! assertion is made an output, so we can simply check the CSV that
 //! souffle emits as in:
 //!
-//!      Q_NAME("dummy_var") :- grounded("dummy_var"), ASSERTION(<ARGS>).
-//!      grounded("dummy_var").
-//!      .output Q_NAME
+//!      // Q_NAME("dummy_var") :- grounded("dummy_var"), ASSERTION(<ARGS>).
+//!      // grounded("dummy_var").
+//!      // .output Q_NAME
```

Comment:
I'm not sure if you're aware of this feature, but Rust likes to make sure that the code that you put in as examples are actually valid Rust:

https://doc.rust-lang.org/rustdoc/documentation-tests.html

It appears to do this whenever there is a Markdown code block, and assumes that any code without a language attached is Rust. Therefore, another way to solve this might be to explicitly put in a language other than Rust, like so:

//! ```Datalog
//!      Q_NAME("dummy_var") :- grounded("dummy_var"), ASSERTION(<ARGS>).
//!      grounded("dummy_var").
//!      .output Q_NAME
//! ```

There is no guarantees that Rustdoc will be aware of Datalog syntax highlighting, but the above would be most correct if it did. If not, any language (is there a text one?) would suppress Rust trying to ensure that the code compiles and runs.

That said, your commenting solution also works, so I'm fine with it as is if you are ok with it currently.

Reply:
OK that works. So it's that way :)